### ISSUE TITLE
Extended types are ignored

### ISSUE THREAD

**Issue by Baloche** [apollographql/eslint-plugin-graphql#227]
Hello 👋

My schema :
```graphql
type Query {
  games: [Game]!
}

type Game {
  id: ID!
  name: String!
}

extend type Game {
  playersCount: Int!
}
```

My query : 
```javascript
const query = gql`
  query games {
    games {
      id
      name
      playersCount
  }
`
```

The warning I get :
```
Cannot query field "playersCount" on type "Game". eslint(graphql/template-strings)
```

Extended types are not supported by the plugin or am I doing something wrong ?

**Comment by jnwng** [comment]
it'd be useful to get more information about the schema you're using — if you wouldn't mind, can you share the actual JSON representation of the schema that you started with?

it'd also be useful to see how you're configuring this plugin in your `.eslintrc` to figure out whether the `extend type` is showing up in the schema to begin with.

**Comment by Baloche** [comment]
I'm sorry, I don't get which informations you need. This is my entire schema defined in a `schema.graphql` file.

Here is my `.eslintrc.js` where `schemaString` is my entire schema as a string : 

```js
module.exports = {
  extends: ['plugin:react/recommended'],
  plugins: ['graphql'],
  rules: {
    'graphql/template-strings': ['error', { env: 'apollo', schemaString }],
    'graphql/required-fields': ['error', { env: 'apollo', schemaString, requiredFields: ['id'] }],
  },
}
```

I don't have myself a JSON representation of the schema.

**Comment by jnwng** [comment]
can you share the schemaString? or is it directly the SDL you shared
already?

**Comment by Baloche** [comment]
Yes, `schemaString` is the schema I already shared, in SDL, as a string.

**Comment by jnwng** [comment]
this took me down a huge rabbit hole and i found an answer.

under the hood, `eslint-plugin-graphql` is just delegating to `graphql-js` to handle schema parsing, we just provide a few options of how you pass that schema to validated your graphql documents. in this case, with the schemaString option, we call [buildSchema](https://github.com/apollographql/eslint-plugin-graphql/blob/ea4e38af420f01359670204f6b36ac00fff41882/src/index.js#L340-L342) from the `graphql` package, which in turn calls [buildASTSchema](https://github.com/graphql/graphql-js/blob/413fc905827f39d2f3e9f93007a2a1e317056107/src/utilities/buildASTSchema.js#L485-L494) internally, which led me to [this open issue](https://github.com/graphql/graphql-js/issues/922) on supporting `extend` within the `buildASTSchema`, which looks like it's slated for release in `graphql@15.0.0`

however, afaik, `graphql-tools` already [supports creating schemas](https://www.apollographql.com/docs/graphql-tools/generate-schema#extend-types) with the `extend` keyword, so if you create the schema using `graphql-tools`, you can send it via the `schemaJson` option and this should all lint properly.

i hope this helps!
